src/lib/orion_websocket/parser.cpp

```diff
@@ -84,4 +87,42 @@ void ws_parser_parse
   {
     head.gotHeaders = false;
   }
+
+  if (doc.HasMember("params") && doc["params"].IsObject())
+  {
+    rapidjson::Value::ConstMemberIterator it = doc["params"].MemberBegin();
+    while (it != doc["params"].MemberEnd())
+    {
+      uriParamGet(ciP, it->name.GetString(), it->value.GetString());
+      ++it;
+    }
+  }
+}
+
+const char *ws_parser_message
+(
+ const std::string&  msg,
+ const HttpHeaders&  head,
+ int                 statusCode
+)
+{
+  const char *tmpl = "{\"headers\": %s, \"message\": %s, \"status\": \"%d\"}";
+
+  rapidjson::StringBuffer buff;
+  rapidjson::Writer<rapidjson::StringBuffer> writer(buff);
+  writer.StartObject();
+  std::map<std::string, std::string *>::const_iterator it = head.headerMap.begin();
+  while (it != head.headerMap.end())
+  {
+    writer.Key(it->first.c_str());
+    writer.String(it->second->c_str());
+    ++it;
+  }
+  writer.EndObject();
+
+  const char *headers = buff.GetString();
+  char *json = (char *) malloc(strlen(tmpl) + strlen(headers) + msg.size() + 1);
+  sprintf(json, tmpl, headers, msg.c_str(), statusCode);
```

Comment:
Yes, my mistake, it will be fixed. But status code has a maximum of 3 digits maybe the best choice is: `strlen(tmpl) - 6 + strlen(headers) + msg.size() + 3 + 1` 
and use a IF statement to discard any code when code < 100 && code > 599